core/services/synchronization/telemetry_ingress_batch_client.go

```diff
@@ -127,14 +130,54 @@ func (tc *telemetryIngressBatchClient) start(ctx context.Context) error {
 			}
 			tc.telemClient = telemPb.NewTelemClient(conn)
 			tc.closeFn = func() error { conn.Close(); return nil }
+			tc.startHealthMonitoring(ctx, conn)
 		}
 	}
 
 	return nil
 }
 
+// startHealthMonitoring starts a goroutine to monitor the connection state and update other relevant metrics every 5 seconds
+func (tc *telemetryIngressBatchClient) startHealthMonitoring(ctx context.Context, conn *wsrpc.ClientConn) {
+	_, cancel := context.WithCancel(ctx)
+	tc.healthMonitorCancel = cancel
+
+	tc.eng.Go(func(ctx context.Context) {
+		ticker := time.NewTicker(5 * time.Second)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				// Check the connection state
+				connected := float64(0)
+				if conn.GetState() == connectivity.Ready {
+					connected = float64(1)
+				}
+				TelemetryClientConnectionStatus.WithLabelValues(tc.url.String()).Set(connected)
+				tc.connected.Store(connected == 1)
+
+				// Report number of workers
+				tc.workersMutex.RLock()
+				TelemetryClientWorkers.WithLabelValues(tc.url.String()).Set(float64(len(tc.workers)))
```

Comment:
IIRC the workers are never spun down after being added, so this can be a counter that's incremented when a worker is created

https://github.com/smartcontractkit/chainlink/blob/develop/core/services/synchronization/telemetry_ingress_batch_client.go#L204